src/main/java/htsjdk/samtools/util/Histogram.java

```diff
@@ -317,14 +320,24 @@ public double getPercentile(final double percentile) {
         if (percentile <= 0) throw new IllegalArgumentException("Cannot query percentiles of 0 or below");
         if (percentile >= 1) throw new IllegalArgumentException("Cannot query percentiles of 1 or above");
 
-        double total = getCount();
+        values().stream()
+                .filter(b -> b.getValue() < 0)
+                .findFirst()
+                .ifPresent(b -> {
+            throw new IllegalArgumentException("Cannot calculate Percentile when negative counts are present " +
+                    "in histogram. Bin " + b.getId() + "=" + b.getValue());
+        });
+
+        final double total = getCount();
```

Comment:
`getCount()` iterates also every value, the same as the check for the negative values. I will recommend for efficiency (this PR adds a third iteration over all the values) to do it in one go:

```java
double total = 0;
for (final Bin<K> b : values()) {
    if (b.getValue() < 0) {
        throw new IllegalArgumentException("Cannot calculate Percentile when negative counts are present " +
                "in histogram. Bin " + b.getId() + "=" + b.getValue());
    }
    total += bin.getValue();
}
```

This will safe an extra iteration...

Reply:
no. since there's an early return that might happen before the iteration concludes....

Reply:
For a call on a histogram with correct values, the check and getCount perform two complete iterations, and it's when it is supposed to be used. The last iteration might return early, but before the values were iterated twice....

Reply:
@yfarjoun I think @magicDGS is suggesting to replace the getCount() call with a for-loop that also checks nothing is negative, not to add the check to the existing for loop that finds the bin, which would potentially terminate to soon.

It seems like a reasonable optimization.  I don't think it's particularly necessary because I don't think there's any software that has a bottleneck on `getPercentile()`, so it may make more sense to leave the checks independent for clarity.

Reply:
the issue is that the Histogram class might be used with negative values...and in those cases getCount should work. Its only in the case that you need to convert a Histogram to a density that you need the values to be non-negative and that the count will be positive...